File: spring-httpclient/src/main/java/com/example/httpclient/component/DefaultApplicationRunner.java
package com.example.httpclient.component;

import org.springframework.boot.ApplicationArguments;
import org.springframework.boot.ApplicationRunner;
import org.springframework.stereotype.Component;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@Component
@RequiredArgsConstructor
public class DefaultApplicationRunner implements ApplicationRunner {

    @Override
    public void run(ApplicationArguments args) throws Exception {
        log.info("## ApplicationRunner");

        var optNames = args.getOptionNames();
        for (var name : optNames) {
            log.info("option: name={}, value={}", name, args.getOptionValues(name));
        }

        var nonOptArgs = args.getNonOptionArgs();
        for (var arg : nonOptArgs) {
            log.info("non option args: {}", arg);
        }

        var srcArgs = args.getSourceArgs();
        for (var arg : srcArgs) {
            log.info("source arg: {}", arg);
        }

    }
}

File: spring-httpclient-kotlin/src/main/kotlin/com/example/httpclient/component/DefaultApplicationRunner.kt
package com.example.httpclient.component

import io.github.oshai.kotlinlogging.KotlinLogging
import org.springframework.boot.ApplicationArguments
import org.springframework.boot.ApplicationRunner
import org.springframework.stereotype.Component

@Component
class DefaultApplicationRunner : ApplicationRunner {
    private val log = KotlinLogging.logger {}

    override fun run(args: ApplicationArguments) {
        log.info { "## ApplicationRunner" }

        args.optionNames.forEach { name ->
            log.info { "option: name=$name, value=${args.getOptionValues(name)}" }
        }

        args.nonOptionArgs.forEach { arg ->
            log.info { "non option args: $arg" }
        }

        args.sourceArgs.forEach { arg ->
            log.info { "source arg: $arg" }
        }
    }
}
